### ISSUE TITLE
[Tagging] Handle import course when the current user does not have permission

### ISSUE THREAD

**Issue by ChrisChV** [openedx/modular-learning#204]
This is a follow-up ticket of the Scenario 4 in https://github.com/openedx/modular-learning/issues/182 
> Scenario 4: Tags in the OLX match a Taxonomy (by Export ID) but the current user does not have permission to view/use that taxonomy [because it's linked to a different org]:
Tags will be imported and visible in Studio. Course author will be able to delete them if they want. Course author will not be able to add additional tags from the same taxonomy though, because they don't have permission.

Currently these types of tags are imported and created without problems, but since the user does not have permissions they do not appear in Studio (Manage tags drawer).

**Acceptance Criteria**

- Course author will be abble to import tags, but if does not have permission to view/use that taxonomy [because it's linked to a different org]: Tags will be imported and visible in Studio
- Course author will be able to delete them if they want. Course author will not be able to add additional tags from the same taxonomy though, because they don't have permission. ([ref](https://github.com/openedx/edx-platform/pull/34356#issuecomment-2021419459))

**Extra information**

- Discussion: https://github.com/openedx/edx-platform/pull/34356#issuecomment-2021332287
- Proof of concept:
    - https://github.com/open-craft/edx-platform/pull/649
    - https://github.com/openedx/frontend-app-course-authoring/pull/926

**Comment by ChrisChV** [comment]
Continue the disscussion here.
https://github.com/openedx/edx-platform/pull/34356#issuecomment-2030545698
> Is there some way we could do it without changing the taxonomy permissions or retrieving "all taxonomies"? Perhaps just list all the objecttags and if any don't match the "allowed" taxonomies, we can display the rest under a heading like "Other Tags"

@bradenmacdonald  Yes, it is possible, easier and cleaner. I just realized this requirement: `Course author will be able to delete them if they want`. In the proof of concept the Course author cannot remove tags. The permissions will have to be modified there.

**Comment by rpenido** [comment]
> The permissions will have to be modified there.
For the backend, we could create a new `can_remove_tags` and check it at the view level if we receive an empty tag list, instead of checking `can_tag_object`.

**Comment by ChrisChV** [comment]
@yusuf-musleh This is my implementation plan:

- Implement a new section in the bottom of the Tag Drawer with "Other Tags" ([ref](https://github.com/openedx/edx-platform/pull/34356#issuecomment-2030545698))
- We can put collapsibles for each taxonomy without making a request for this taxonomies because the content tags has the basic data of taxonomy (name). 
- Use the same styles and components of the Drawer.
- If the user dosen't have permission, then the taxonomy will appear in "Other Tags", otherwise will appear in normal section. So no need to make any permission changes here.
- The collapsibles in "Other Tags" they will not have the select component to add tags.
- [Update the permission](https://github.com/openedx/openedx-learning/blob/main/openedx_tagging/core/tagging/rest_api/v1/serializers.py#L143) to verify if the user can remove the object tag ([ref](https://github.com/openedx/modular-learning/issues/204#issuecomment-2034663591))

@yusuf-musleh @bradenmacdonald @ali-hugo @jmakowski1123 What do you think?

**Comment by ali-hugo** [comment]
@ChrisChV I decided to create two quick mock ups to make sure I'm interpreting this correctly. Please let me know whether _Option 1_ or _Option 2_ better reflects what you have in mind (and if I've captured all the details correctly).

### Option 1:
![Option 1](https://github.com/openedx/modular-learning/assets/88187719/8a72e13a-1297-4113-b7aa-8e9b9188e9fd)

### Option 2:
![Option 2](https://github.com/openedx/modular-learning/assets/88187719/0340d8bf-fd4a-4cf0-9046-5744fd28a3fc)

**Comment by yusuf-musleh** [comment]
@ChrisChV I like the plan you mentioned and I think its a good approach. 

I feel @ali-hugo 's option 2 might be more clear and less confusing for the user.

**Comment by bradenmacdonald** [comment]
@ChrisChV Sounds good! And tahnks @ali-hugo for mocking this up.

I think this may be confusing for the user in any case, so maybe we should add a message to the Other Tags section? Something like "These tags are already applied, but their taxonomies aren't available for use so you cannot add new tags at this time."

**Comment by ChrisChV** [comment]
> "These tags are already applied, but their taxonomies aren't available for use so you cannot add new tags at this time."

:+1:  to this

@ali-hugo Thanks for the mockups. I think the second mockup is the best, but it will take longer, since I would have to place collapsibles inside the "Other tags" collapsible. I was thinking something like this:

![image](https://github.com/openedx/modular-learning/assets/11827305/be2ff29e-2691-45cf-a7ed-cb7638812399)

- Move "Other tags" to the left as a header like "Mange tags" (not a collapsible)
- Move taxonomies to the the left as a normal taxonomy collapsibles.
- Not show "Other tags" if there are no tags of that type

**Comment by ali-hugo** [comment]
@ChrisChV @bradenmacdonald Thanks! I've incorporated your feedback into the mockup below. Let me know if there's anything else you think we should change. 

P.s. I tweaked @bradenmacdonald's message text slightly; I'm trying make it clearer to the user why they can't use the taxonomies. Let me know if you think it could be improved.

![Other tags](https://github.com/openedx/modular-learning/assets/88187719/24c2df82-95a9-464a-a636-295460eb15f5)
